### DISCUSSION TITLE
amplify with interactive value

### DISCUSSION THREAD

**Discussion by YoannQueret** [savonet/liquidsoap#3352] [Q&A] [answered]
Hi,

opam : 4.12.0
liquidsoap : 2.1.4
os : debian 12

Is it possible to set amplify from interactive value ?

The code below, return an error, and i don't see an other way to do that :

```
volume = interactive.float("volume", 1.)
source = input....
source = amplify(lin_of_dB(volume),source)
```

```

Error 5: this value has type
  () -> _ (inferred at /home/yoann/.opam/4.12.0/lib/liquidsoap/share/liquidsoap/2.1.4/libs/interactive.liq, line 425, char 2-107)
but it should be a subtype of the type of the value at /home/yoann/.opam/4.12.0/lib/liquidsoap/share/liquidsoap/2.1.4/libs/math.liq, line 21, char 14
  float (inferred at /home/yoann/.opam/4.12.0/lib/liquidsoap/share/liquidsoap/2.1.4/libs/math.liq, line 22, char 15-18)
```

**Comment by vitoyucepi** [comment, marked as answer]
Hi @YoannQueret,

1. `volume = interactive.float("volume", 1.)`
   `volume` is a function that will return the value of the interactive element.
   You should call it like this `volume()`.
2. To use `lin_of_dB(volume)`, you need to pass a float value instead of a function.
   `lin_of_dB(0.5)`, `lin_of_dB(10.)` and `lin_of_dB(volume())`.
3. `amplify` expects a function as the first argument that returns a float.
   Float will be fine too, but there's a catch.
   The value of `amplify(lin_of_dB(volume()), source)` will be evaluated only once at the startup.
   So, to make it a function, you can construct an anonymous function using `{}`.
   It will look like `amplify({lin_of_dB(volume())}, source)`.

So, the result is
```ruby
volume = interactive.float("volume", 1.)
source = ...
source = amplify({lin_of_dB(volume())}, source)
```

**Reply by YoannQueret** [reply]
Hi @vitoyucepi 

Thk a lot for your quick answer and solution.

I don't known the tips about the anonymous function ! Very usefull !
